src/vivarium_gates_iv_iron/components/pregnancy.py

```diff
@@ -228,9 +233,11 @@ def _sample_initial_pregnancy_state_change_date(
     def _sample_pregnancy_outcome_and_duration(
```

Comment:
I'm realizing that I may have set us up for failure here. I think I've instructed to determine pregnancy outcome at the start of pregnancy so that we can determine duration, probability of stillbirth should really depend on hemoglobin level at the end of pregnancy, not the beginning.

I think to get around this, we can determine if someone will have an "other" outcome OR a live/still birth at the start of pregnancy and determine the duration like normal. Then at the time of birth, use the value modifiers to determine if it is a live or still birth.

Sorry, I think this was a defect in my documentation. Essentially, if we determine the risk of stillbirth based on hemoglobin levels at the beginning of pregnancy then it won't be affected by our interventions which increase hemoglobin levels in the middle of pregnancy.

If this is what you've already done then that's awesome, but otherwise let me know if you want to chat about it!